implementations/rust/ockam/ockam_command/src/enroll/command.rs

```diff
@@ -0,0 +1,371 @@
+    }
+}
+
+/// Use an embedded node to enroll a user with a token
+pub async fn enroll(
+    ctx: &Context,
+    opts: &CommandGlobalOpts,
+    token: Auth0Token,
+) -> miette::Result<()> {
+    let node_name = start_embedded_node(ctx, opts, None).await?;
+    enroll_with_node(ctx, opts, &CloudOpts::route(), node_name.as_str(), token).await?;
+    retrieve_user_project(ctx, opts, &node_name)
+        .await
+        .map(|_| ())
+        .into_diagnostic()
+}
+
+async fn default_space<'a>(
+    ctx: &Context,
+    opts: &CommandGlobalOpts,
+    node_name: &str,
+) -> Result<Space<'a>> {
+    // Get available spaces for node's identity
+    opts.terminal
+        .write_line(&fmt_log!("Getting available spaces in your account..."))?;
+
+    let mut rpc = RpcBuilder::new(ctx, opts, node_name).build();
+    let is_finished = Mutex::new(false);
+    let send_req = async {
+        rpc.request(api::space::list(&CloudOpts::route())).await?;
+        *is_finished.lock().await = true;
+        rpc.parse_response::<Vec<Space>>()
+    };
+
+    let message = vec![format!("Checking for any existing spaces...")];
+    let progress_output = opts.terminal.progress_output(&message, &is_finished);
+
+    let (mut available_spaces, _) = try_join!(send_req, progress_output)?;
+
+    // If the identity has no spaces, create one
+    let default_space = if available_spaces.is_empty() {
+        opts.terminal
+            .write_line(&fmt_para!("No spaces are defined in your account."))?
+            .write_line(&fmt_para!(
+                "Creating a trial space for you ({}) ...",
+                "everything in it will be deleted in 15 days"
+                    .to_string()
+                    .color(OckamColor::FmtWARNBackground.color())
+            ))?
+            .write_line(&fmt_para!(
+            "To learn more about production ready spaces in Ockam Orchestrator, contact us at: {}",
+            "hello@ockam.io".to_string().color(OckamColor::PrimaryResource.color())
+        ))?;
+
+        let is_finished = Mutex::new(false);
+        let name = crate::util::random_name();
+        let mut rpc = RpcBuilder::new(ctx, opts, node_name).build();
+        let send_req = async {
+            let cmd = crate::space::CreateCommand {
+                name: name.to_string(),
+                admins: vec![],
+            };
+
+            rpc.request(api::space::create(&cmd)).await?;
+            *is_finished.lock().await = true;
+            rpc.parse_response::<Space>()
+        };
+
+        let message = vec![format!(
+            "Creating space {}...",
+            name.to_string().color(OckamColor::PrimaryResource.color())
+        )];
+        let progress_output = opts.terminal.progress_output(&message, &is_finished);
+        let (space, _) = try_join!(send_req, progress_output)?;
+        space.to_owned()
+    }
+    // If it has, return the first one on the list
+    else {
+        for space in &available_spaces {
```

Comment:
This will be sensitive to whatever order is returned by the backend, correct?

Reply:
Yes, there seems to be an implicit assumption that the order of spaces, hence the default one, will always be the same.

Reply:
Yes I'm afraid. We could derive the default space from the default project. But if don't have either, we decided to just get the first. The user can later set the default with the `space default` command